cmd/lets-go-chat/main.go

```diff
@@ -1,36 +1,46 @@
 package main
 
 import (
-	"encoding/json"
+	"net/http"
+	"strconv"
+	"log"
+	"flag"
+	"database/sql"
+	"fmt"
 	"github.com/gorilla/mux"
+	_ "github.com/lib/pq"
 	"lets-go-chat/configs"
 	"lets-go-chat/internal/handlers"
-	"log"
-	"net/http"
-	"os"
-	"strconv"
+	rep "lets-go-chat/internal/repositories"
 )
 
-var configuration configs.Configuration
-
 func main() {
-	err := loadConfiguration()
+
+	configFile := flag.String("config", "config.json", "Configuration file in JSON-format")
+	flag.Parse()
+
+	config, err := configs.LoadConfig(*configFile)
 	if err != nil {
-		log.Fatal("CANT LOAD CONFIG")
+		log.Fatal("can't load configuration")
 	}
+
+	psqlconn := fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", config.DBConfig.DBHost, config.DBConfig.DBPort, config.DBConfig.DBUser, config.DBConfig.DBPassword, config.DBConfig.DBName)
+	db, err := sql.Open("postgres", psqlconn)
+	if err != nil {
+		log.Fatal("can't connect to DB")
+	}
+
+	rep.RegisterUserRepository(rep.NewUsersDataRepository(db))
+	defer db.Close()
```

Comment:
it makes sense close db with defer right after error checking because  it's one semantic part